orchestrator/src/main/java/eu/arrowhead/core/orchestrator/OrchestratorController.java

```diff
@@ -256,6 +344,41 @@ private void checkOrchestratorFormRequestDTO(final OrchestrationFormRequestDTO r
 		}
 	}
 	
+	//-------------------------------------------------------------------------------------------------	
+	private void checkOrchestratorStoreFlexibleRequestDTOList(final List<OrchestratorStoreFlexibleRequestDTO> dtoList, final String origin) {
```

Comment:
I suggest additional checks:

- if a system name is specified (consumer and/or provider), you should check its format with the new CommonNamePartVerifier (see SR add system)
- for the service definition if use_strict_service_definition_verifier=true in the config file (this one is not present currently in the orchestrator), you should do the same check
- for the service interface you should check it with its dedicated verifier

I forgot the use the new verifier in the old store methods when I implemented the same functionality in the SR, but I think we should use the verifier there too.

Reply:
Regarding to the old store we assume that systems, serviceDefinitions, interfaces are already in the database. Only in case of foreign providers we will create new records, but we cannot know whether the neighbor cloud is in strict mode or not.

Should it still be tested?

Reply:
You are right, maybe that is the cause why I don't implement it in the first place. :)

But in the case of flexible store, we should check the format, because you can create rules for systems and services that not exist yet.